### DISCUSSION TITLE
How to deal with missing values in accelerometer data

### DISCUSSION THREAD

**Discussion by adeline95** [bmcclintock/momentuHMM#160] [Q&A] [answered]
Hi,
I am a master student at the University of Zurich and I am working on accelerometer data of roe deer fawns with HMM’s with the package momentuHMM.
My main goal is to find out how long it takes for the mother to reunite with its offspring and if it’s possible to use HMM’s on ACC data to do so. More generally, I’m interested in identifying behavioral classes displayed by roe deer fawns and how external/environmental variables (e.g. weather/duration since handling/type of vegetation) affect these behaviors.
• the tags record bursts of 11 seconds out of 20 seconds (non-continuous recording --> 11s recording – 9s no recording), with a frequency of 10Hz for each axis.
• To reduce bursts with mixed behaviours, we summarized them in 1s.
• We inserted NA’s for missing data (9s gaps)  and for gap due to GPS fixes (every 2 minutes or longer, with an irregular duration).

How should we deal with the non-continuous recording and data irregularities (i.e. NA’s)?  NA's could be problematic for the HMM’s?

I thank you in advance for all the suggestion.

Kind regards,

Adeline

**Comment by bmcclintock** [comment, marked as answer]
I'm afraid I can't provide much advice here without a better understanding of the objective(s) and the data. Missing data can be problematic in HMMs; a few here and there are usually OK but it sounds like you have a lot of NAs breaking up the time series. This problem often occurs with telemetry location data and you can find some suggestions in that context by searching for [missing data](https://github.com/bmcclintock/momentuHMM/discussions?discussions_q=%22missing+data%22) in [Discussions](https://github.com/bmcclintock/momentuHMM/discussions). 

If the time scale for the mother to reunite with its offspring is on the order of minutes or hours, then you might be able to arrive at some sort of coarser summary (e.g. by 1 minute, 5 minutes, 20 minutes, 30 minutes, 1 hour, etc.) that would effectively remove the missing data. If there is sufficient serial correlation in the hidden process relative to the time scale of the observations (e.g. state-switching occurs every few hours and the coarser observations are, say, every 15 minutes) and the data are missing more-or-less at random, then this might be a way to work around the issue.

Alternatively, it could be worth exploring continuous-time HMMs. These are now implemented in the ['develop'](https://github.com/bmcclintock/momentuHMM/tree/develop) version of {momentuHMM} via the `fitCTHMM` function. Assuming your accelerometer observations are "instantaneous", i.e., each observation $x_t$ only depends on the state active at time $t$ (rather than the entire state sequence between times $t-1$ and $t$ (e.g., see [Glennie et al. 2022](https://doi.org/10.1111/2041-210X.13801)), then you would only include the (irregular) time steps where you have observations (and forget about the NAs). However, you'd still want sufficient serial correlation in the hidden process relative to the time scale of the observations and the data to be missing more-or-less at random.
